src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+    let currentSeparator;
+
+    if (separator === undefined) {
+      currentSeparator = ',';
```

Comment:
you can get rid of the extra `if` construct here, if you add a default `separator` value

`function(separator = ',')`